architecture/evm/json_rpc_cache.go

```diff
@@ -573,10 +573,15 @@ func (c *EvmJsonRpcCache) getFinalityState(ctx context.Context, req *common.Norm
 
 	blockRef, blockNumber, _ := ExtractBlockReferenceFromRequest(ctx, req)
 
-	if blockRef == "latest" || blockRef == "pending" {
+	// Decide finality from the block reference string:
+	//   • "*" wildcard entry: unfinalized (matches default policies)
+	//   • tag  (first char not a digit): realtime
+	if blockRef == "*" {
```

Comment:
Good catch! If we don't maybe let's add unit tests for hash-based eth_getBlockByNumber?

Reply:
I think you meant `eth_getBlockByHash`?

Because `eth_getBlockByNumber` does not seem to accept hash of the block as param

```
{
    "method": "eth_getBlockByNumber",
    "params": [
        "0x3f07a9c83155594c000642e7d60e8a8a00038d03e9849171a05ed0e2d47acbb3",
        false
    ],
    "id": 1,
    "jsonrpc": "2.0"
}
```

Reply:
Having eth_getBlockByHash for sure is also good to have. I actually meant hex-based block by number (vs tag-based).

Reply:
oh yeah I think for that we have unit-tests

Reply:
Thinking about this again will this be a breaking-change/different behavior than current logic?

e.g. if ref = * but block num is of a very old block, it must be considered FInalized

Reply:
hmm yeah based on the logic we have `*` means I couldn't figure out the block. 

I think the reason we cannot rely on the `resp != nil` to handle that flow is because in `Get` cache, we are not relying on response, so we might be detecting the wrong policy?

https://github.com/erpc/erpc/blob/1016e7d7b4b8ee0be0db64058aaa3ab9a93dc664/architecture/evm/json_rpc_cache.go#L104

Reply:
Yeah that makes sense. Ultimate test would be cases for "cache transaction trace" or "cache eth_getTransactionByHash" where there's no block number in req nor resp.

Reply:
am I correct that for both `trace_transaction` and `eth_getTransactionByHash`, we actually return `blockNumber` as part of response?

https://www.quicknode.com/docs/ethereum/trace_transaction
https://www.quicknode.com/docs/ethereum/eth_getTransactionByHash

Reply:
Ah you're right was thinking about "trace_replayTransaction" trace func where no blockNumber exists.

Re eth_getTransactionByHash it's an interesting case, we can return '*' (as in it's not necessary to invalidate on reorg) but since we'd have a blockNumber probably better to have that as blockRef.

Reply:
so probably safe to remove `ReqRefs: ArbitraryBlock` for both `eth_getTransactionReceipt` and `eth_getTransactionByHash` as both return `blockNumber` as part of their response?

https://github.com/erpc/erpc/blob/1016e7d7b4b8ee0be0db64058aaa3ab9a93dc664/common/defaults.go#L361

Reply:
I believe that's needed for cache hits (i.e. the * means to use the secondary index where blockNum is not needed to 'look up' the cache entry).